rcl/src/rcl/subscription.c

```diff
@@ -272,6 +272,41 @@ rcl_take(
   return RCL_RET_OK;
 }
 
+rcl_ret_t
+rcl_take_raw(
+  const rcl_subscription_t * subscription,
+  rcl_message_raw_t * raw_message,
+  rmw_message_info_t * message_info)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(subscription, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
```

Comment:
It looks like a few of these checks (`subscription`, `subscription->impl`, subscription options, `subscription->impl->rmw_handle`) can be replaced with with a call `rcl_subscription_is_valid()`.